added warning modals to unsaved bounty and profile pages

deeayeen opened the pull request.

corwinharrell commented:
@deeayeen thanks for submitting this! We will take some time to review and will leave any feedback that might need addressing here before we accept your submission and approve your payout.

@mbeylin If you have some time and don't mind, I'd love a second pair of eye's in reviewing this and doing some QA. Let me know if you have any thoughts around anything in particular.

corwinharrell commented:
@deeayeen We have another Bounties Network community member who also applied to this bounty who has asked if it would be ok to contribute to reviewing the pull-request (just as a learning opportunity) as he is currently sharpening his react skillset. Do you mind the extra set of eyes?

deeayeen commented:
Sure, let me know if anything needs any explanation, the react-router can
get confusing.

On Tue, Jul 2, 2019 at 12:25 AM Corwin Harrell <notifications@github.com>
wrote:

> @deeayeen <https://github.com/deeayeen> We have another Bounties Network
> community member who also applied to this bounty who has asked if it would
> be ok to contribute to reviewing the pull-request (just as a learning
> opportunity) as he is currently sharpening his react skillset. Do you mind
> the extra set of eyes?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Bounties-Network/Explorer/pull/373?email_source=notifications&email_token=AIQHEDC35ICWIFIDKR4OJE3P5JY4ZA5CNFSM4H4SWZSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY7MWHI#issuecomment-507431709>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIQHEDAXO37J4RF2OXTQ6C3P5JY4ZANCNFSM4H4SWZSA>
> .
>

corwinharrell commented:
@deeayeen awesome, thanks for being flexible!

@adnfx2 feel free to jump in 😄

adnfx2 commented:
Thanks guys for being so nice and letting me participate here. 
Here I go:

1) In CreateBounty you defined the following states: dirty, isit and showModal, I see the point of creating dirty, and it's also being used down there. But isit and showModal are not being used anywhere in the code, same thing happened with the following methods defined in this class: showModal, hideModal and hideModalTemp. So my question is why are they defined in first place? Are they really needed?

2) I tested the code a bit and I found the following behaviors:
- In CreateBounty form, when clicking on: - create bounty or create draft or update draft - to submit the form, the modal is displayed, I'm guessing this should not happen(Please correct if I'm wrong).
- In Settings when editing EmailPreferences, no modal is displayed if I attempt to leave, only when I modified UserSettings the modal will be potentially displayed. Also in UserSettings if I make an edition and successfully gets saved after clicking update profile, when attempting to leave that page, a modal will be displayed, this prolly is not the intended behavior.

Hope that my observations can become useful to you guys, or if you think I might have done something wrong when testing, please let me know.

deeayeen commented:
Perfect, that is very useful feedback Adrian, thank you.

Corwin:
To sum up
1) I declared some variables during testing that I need to delete.
2) I am currently blocking all routing if the form has unsaved changes,
including the submit form route. I need to whitelist certain routes from
being blocked.
3) Only added modal to user settings, need to add modal when editing email
preferences as well.

I will submit these fixes shortly.

On Tue, Jul 2, 2019 at 1:35 AM Adrian Pirela <notifications@github.com>
wrote:

> Thanks guys for being so nice and letting me participate here.
> Here I go:
>
>    1.
>
>    In CreateBounty you defined the following states: dirty, isit and
>    showModal, I see the point of creating dirty, and it's also being used down
>    there. But isit and showModal are not being used anywhere in the code, same
>    thing happened with the following methods defined in this class: showModal,
>    hideModal and hideModalTemp. So my question is why are they defined in
>    first place? Are they really needed?
>    2.
>
>    I tested the code a bit and I found the following behaviors:
>
>
>    - In CreateBounty form, when clicking on: - create bounty or create
>    draft or update draft - to submit the form, the modal is displayed, I'm
>    guessing this should not happen(Please correct if I'm wrong).
>    - In Settings when editing EmailPreferences, no modal is displayed if
>    I attempt to leave, only when I modified UserSettings the modal will be
>    potentially displayed. Also in UserSettings if I make an edition and
>    successfully gets saved after clicking update profile, when attempting to
>    leave that page, a modal will be displayed, this prolly is not the intended
>    behavior.
>
> Hope that my observations can become useful to you guys, or if you think I
> might have done something wrong when testing, please let me know.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Bounties-Network/Explorer/pull/373?email_source=notifications&email_token=AIQHEDFTKGYOOAVLCHGBNLLP5KBCNA5CNFSM4H4SWZSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY7RDFI#issuecomment-507449749>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIQHEDH32BLGCMT4KPHLVSTP5KBCNANCNFSM4H4SWZSA>
> .
>

adnfx2 commented:
Hey man that's awesome!, one more thing, the content of the modal like it's title: "Discard Changes?" it's hard coded, I think you should use intl.get to place the content there from us_EN.json

deeayeen commented:
Submitted updated PR.

On Tue, Jul 2, 2019 at 12:33 AM Corwin Harrell <notifications@github.com>
wrote:

> @deeayeen <https://github.com/deeayeen> awesome, thanks for being
> flexible!
>
> @adnfx2 <https://github.com/adnfx2> feel free to jump in 😄
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Bounties-Network/Explorer/pull/373?email_source=notifications&email_token=AIQHEDHJKW62OH7BV3VIOFLP5JZ4FA5CNFSM4H4SWZSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY7NJEA#issuecomment-507434128>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIQHEDCHQZQPTU7PZ3UHRV3P5JZ4FANCNFSM4H4SWZSA>
> .
>

mbeylin commented:
I have some notes on this PR:
- When editing an existing bounty, I can click the "increase payout" link to be taken to the bounty detail page. Instead of seeing our custom modal, I'm presented with this one instead:
![Screen Shot 2019-07-02 at 1 11 24 PM](https://user-images.githubusercontent.com/6082639/60532344-1a986580-9ccb-11e9-8d4b-48f7a9743709.png)
This should be replaced with our custom modal.
- Some of the if statements you've added are quite messy. For instance:
```
        if (
	      this.state.dirty &&
	      !isEditing &&
	      (this.state.dirty && !isDraft) &&
	      (this.state.dirty && this.state.submitNotPressed)
	    )
``` 

should be simplified to:
```if (this.state.dirty &&  !isEditing && !isDraft && this.state.submitNotPressed)```
Please simplify this and other if statements in the codebase to maintain simplicity and efficiency.
- There's still a variable `handleUser` from UserSettings.js that is assigned a value and never used.

mbeylin commented:
To test out the editing of bounties, you can use the command `yarn start-rinkeby` and run a version of the explorer that points to the Rinkeby test network, on which you can easily create a new bounty without having to pay for gas on the main net.

corwinharrell commented on src/containers/CreateBounty/index.js:
```diff
@@ -64,23 +71,145 @@ class CreateBountyComponent extends React.Component {
+                  onClose={onCancel}
+                >
+                  <Modal.Header closable />
+                  <Modal.Body>
```
For our modal component, formatting is built in, and you shouldn't need to add any additional classNames or styles. For reference, take a look at `src/containers/Bounty/components/modals/ContributeFormModal.js` to see how `Modal.Message` and `Modal.Description` are used to determine how different elements of the modal are structured and styled. Please remove any inline styles and HTML tags and defer to composing the modal in this way.

As others has also mentioned, please replace the text content with copy added to `src/i18n/en_US.json`.

corwinharrell commented on src/containers/CreateBounty/index.js:
```diff
@@ -64,23 +71,145 @@ class CreateBountyComponent extends React.Component {
+                  onClose={onCancel}
+                >
+                  <Modal.Header closable />
+                  <Modal.Body>
```
For modal formatting, see comments above.

deeayeen commented:
Submitted a PR with the following changes:

- Removed unused variables
- Improved if statements
- Corrected modal formatting
- Internationalized modal text

When creating a new bounty on the Rinkeby network it loads infinitely.

On Wed, Jul 3, 2019 at 7:59 PM Corwin Harrell <notifications@github.com>
wrote:

> *@corwinharrell* requested changes on this pull request.
> ------------------------------
>
> In src/containers/CreateBounty/index.js
> <https://github.com/Bounties-Network/Explorer/pull/373#discussion_r300063718>
> :
>
> > +            !nextLocation ||
> +            !nextLocation.pathname.startsWith(crntLocation.pathname)
> +          }
> +        >
> +          {({ isActive, onCancel, onConfirm }) => {
> +            if (isActive) {
> +              return (
> +                <Modal
> +                  dismissable
> +                  size={'medium'}
> +                  fixed
> +                  visible={true}
> +                  onClose={onCancel}
> +                >
> +                  <Modal.Header closable />
> +                  <Modal.Body>
>
> For our modal component, formatting is built in, and you shouldn't need to
> add any additional classNames or styles. For reference, take a look at
> src/containers/Bounty/components/modals/ContributeFormModal.js to see how
> Modal.Message and Modal.Description are used to determine how different
> elements of the modal are structured and styled. Please remove any inline
> styles and HTML tags and defer to composing the modal in this way.
>
> As others has also mentioned, please replace the text content with copy
> added to src/i18n/en_US.json.
> ------------------------------
>
> In src/containers/CreateBounty/index.js
> <https://github.com/Bounties-Network/Explorer/pull/373#discussion_r300063859>
> :
>
> > +            !nextLocation ||
> +            !nextLocation.pathname.startsWith(crntLocation.pathname)
> +          }
> +        >
> +          {({ isActive, onCancel, onConfirm }) => {
> +            if (isActive) {
> +              return (
> +                <Modal
> +                  dismissable
> +                  size={'medium'}
> +                  fixed
> +                  visible={true}
> +                  onClose={onCancel}
> +                >
> +                  <Modal.Header closable />
> +                  <Modal.Body>
>
> For modal formatting, see comments above.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Bounties-Network/Explorer/pull/373?email_source=notifications&email_token=AIQHEDB23TN5TKUTP2U5OTDP5TLF5A5CNFSM4H4SWZSKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB5NP26Q#pullrequestreview-257621370>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIQHEDE5X4BPZDNVSCAW3O3P5TLF5ANCNFSM4H4SWZSA>
> .
>

deeayeen commented:
It seems as though the “increase funds” is an external redirect,
react-router has no way to catch these kinds of redirects (refreshing the
page, entering a new url, etc) so I integrated the pop up prompt you
mentioned for these special cases. I hope this is alright. Other than that
I have addressed all issues you guys have mentioned, please let me know
what else I can help with.

On Fri, Jul 5, 2019 at 2:49 PM Dan Halperin <dee.aye.en@gmail.com> wrote:

> Submitted a PR with the following changes:
>
> - Removed unused variables
> - Improved if statements
> - Corrected modal formatting
> - Internationalized modal text
>
> When creating a new bounty on the Rinkeby network it loads infinitely.
>
> On Wed, Jul 3, 2019 at 7:59 PM Corwin Harrell <notifications@github.com>
> wrote:
>
>> *@corwinharrell* requested changes on this pull request.
>> ------------------------------
>>
>> In src/containers/CreateBounty/index.js
>> <https://github.com/Bounties-Network/Explorer/pull/373#discussion_r300063718>
>> :
>>
>> > +            !nextLocation ||
>> +            !nextLocation.pathname.startsWith(crntLocation.pathname)
>> +          }
>> +        >
>> +          {({ isActive, onCancel, onConfirm }) => {
>> +            if (isActive) {
>> +              return (
>> +                <Modal
>> +                  dismissable
>> +                  size={'medium'}
>> +                  fixed
>> +                  visible={true}
>> +                  onClose={onCancel}
>> +                >
>> +                  <Modal.Header closable />
>> +                  <Modal.Body>
>>
>> For our modal component, formatting is built in, and you shouldn't need
>> to add any additional classNames or styles. For reference, take a look at
>> src/containers/Bounty/components/modals/ContributeFormModal.js to see
>> how Modal.Message and Modal.Description are used to determine how
>> different elements of the modal are structured and styled. Please remove
>> any inline styles and HTML tags and defer to composing the modal in this
>> way.
>>
>> As others has also mentioned, please replace the text content with copy
>> added to src/i18n/en_US.json.
>> ------------------------------
>>
>> In src/containers/CreateBounty/index.js
>> <https://github.com/Bounties-Network/Explorer/pull/373#discussion_r300063859>
>> :
>>
>> > +            !nextLocation ||
>> +            !nextLocation.pathname.startsWith(crntLocation.pathname)
>> +          }
>> +        >
>> +          {({ isActive, onCancel, onConfirm }) => {
>> +            if (isActive) {
>> +              return (
>> +                <Modal
>> +                  dismissable
>> +                  size={'medium'}
>> +                  fixed
>> +                  visible={true}
>> +                  onClose={onCancel}
>> +                >
>> +                  <Modal.Header closable />
>> +                  <Modal.Body>
>>
>> For modal formatting, see comments above.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Bounties-Network/Explorer/pull/373?email_source=notifications&email_token=AIQHEDB23TN5TKUTP2U5OTDP5TLF5A5CNFSM4H4SWZSKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB5NP26Q#pullrequestreview-257621370>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AIQHEDE5X4BPZDNVSCAW3O3P5TLF5ANCNFSM4H4SWZSA>
>> .
>>
>

corwinharrell commented:
@mbeylin any thoughts on the above?
